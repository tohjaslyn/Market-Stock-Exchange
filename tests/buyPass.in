add trader1 100
add trader2 100
import trader2 ABC 10
sell trader2 ABC 10 2
sell trader2 ABC 5 2.50
buy trader1 ABC 10 3
import trader2 DEF 10
buy trader1 DEF 10 3
sell trader2 DEF 10 2
sell trader2 DEF 5 2.50
import trader2 GHI 20
sell trader2 GHI 12 2
buy trader1 GHI 10 3
exit
