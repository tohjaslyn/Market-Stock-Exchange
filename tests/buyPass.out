$ Success.
$ Success.
$ Trader now has 10.00 units of ABC.
$ No trades could be made, order added to sell book.
$ Order could not be placed onto the market.
$ Product bought in entirety, trades as follows:
trader2->trader1: 10.00xABC for $2.00.
$ Trader now has 10.00 units of DEF.
$ No trades could be made, order added to buy book.
$ Product sold in entirety, trades as follows:
trader2->trader1: 10.00xDEF for $3.00.
$ Order could not be placed onto the market.
$ Trader now has 20.00 units of GHI.
$ No trades could be made, order added to sell book.
$ Product bought in entirety, trades as follows:
trader2->trader1: 10.00xGHI for $2.00.
$ Have a nice day.
